src/connection/index.ts

```diff
@@ -31,29 +38,149 @@ export class Connection {
     );
   }
 
+  private async getSytemEngineEndpoint(): Promise<string> {
+    const { apiEndpoint, httpClient } = this.context;
+    const accountName = this.options.account;
+    const url = `${apiEndpoint}/${ACCOUNT_SYSTEM_ENGINE(accountName)}`;
+    const { engineUrl } = await httpClient
+      .request<{ engineUrl: string }>("GET", url)
+      .ready();
+    return engineUrl;
+  }
+
+  private async isDatabaseAccessible(databaseName: string): Promise<boolean> {
+    const query =
+      "SELECT database_name FROM information_schema.databases " +
+      `WHERE database_name='${databaseName}'`;
+
+    const statement = await this.execute(query);
+    const { data } = await statement.fetchResult();
+    return data.length == 1;
+  }
+
+  private async getEngineUrl(
+    engineName: string,
+    databaseName: string
+  ): Promise<string> {
+    const query =
+      "SELECT engs.url, engs.attached_to, dbs.database_name, engs.status " +
+      "FROM information_schema.engines as engs " +
+      "LEFT JOIN information_schema.databases as dbs " +
+      "ON engs.attached_to = dbs.database_name " +
+      `WHERE engs.engine_name = '${engineName}'`;
+    const statement = await this.execute(query);
+    const { data } = await statement.fetchResult();
+    if (data.length == 0) {
+      throw new ConnectionError({ message: `Engine ${engineName} not found.` });
+    }
+    const filteredRows = [];
+    for (const row of data) {
+      if ((row as string[])[2] == databaseName) {
```

Comment:
lets destructure all columns like this `const [url, attached_to, database_name, status] = row` and use constants below instead of indexes

Reply:
Yeah, makes sense. Didn't realise this syntax was available while working on the change.